### PR TITLE
chore: move anthropic as optional

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

Describe your changes in detail.

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by WHALEEYE — APPROVED** [review]
LGTM

File: camel/configs/anthropic_config.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 from typing import List, Union
 
-from anthropic import NOT_GIVEN, NotGiven
+from openai._types import NOT_GIVEN, NotGiven
```
Comment (lightaime) [inline camel/configs/anthropic_config.py:18]: Import `openai` for `anthropic` types seems weird. Can we import `anthropic` inside the class instead?

File: camel/models/anthropic_model.py [hunk 2]
```diff
@@ -54,6 +54,8 @@ def __init__(
         url: Optional[str] = None,
         token_counter: Optional[BaseTokenCounter] = None,
     ) -> None:
+        from anthropic import Anthropic
```
Comment (lightaime) [inline camel/models/anthropic_model.py:57]: Add dependencey requried decorator.

File: camel/utils/token_counting.py [hunk 3]
```diff
@@ -220,6 +219,7 @@ def _count_tokens_from_image(
 class AnthropicTokenCounter(BaseTokenCounter):
     def __init__(self):
         r"""Constructor for the token counter for Anthropic models."""
+        from anthropic import Anthropic
```
Comment (lightaime) [inline camel/utils/token_counting.py:222]: Add dependencey requried decorator.

**Pull request merged**